integration/securitypolicy_test.go

```diff
@@ -0,0 +1,400 @@
+package integration
+
+import (
+	"context"
+	"testing"
+
+	"github.com/TykTechnologies/tyk-operator/api/v1alpha1"
+	"github.com/TykTechnologies/tyk-operator/controllers"
+	tykClient "github.com/TykTechnologies/tyk-operator/pkg/client"
+	"github.com/TykTechnologies/tyk-operator/pkg/client/klient"
+	"github.com/matryer/is"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	cr "sigs.k8s.io/controller-runtime/pkg/client"
+	util "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestSecurityPolicyMigration(t *testing.T) {
+	const (
+		opNs                  = "tyk-operator-system"
+		initialK8sPolicyTag   = "sample-tag"
+		initialK8sPolicyRate  = 50
+		initialK8sPolicyState = "deny"
+	)
+
+	spec := v1alpha1.SecurityPolicySpec{
+		Name:   "existing-spec",
+		State:  "draft",
+		Rate:   34,
+		Active: true,
+		Tags:   []string{"testing"},
+	}
+
+	hasSameValues := func(k8sSpec, tykSpec *v1alpha1.SecurityPolicySpec, k8sStatusID string) bool {
```

Comment:
`k8sStatusID` is not used

Reply:
uh, thank you for spotting it. i totally missed that. updated accordingly in d17880a1cd921024b5b82813d7c62c2021ebb2cd